apps/cow-fi/components/SearchBar/index.tsx

```diff
@@ -79,53 +87,42 @@ export const SearchBar: React.FC<SearchBarProps> = ({ articles }) => {
   useEffect(() => {
     const trimmedQuery = debouncedQuery.trim()
 
-    if (trimmedQuery.length > 0 && trimmedQuery.length < MIN_SEARCH_LENGTH) {
+    if (trimmedQuery.length === 0) {
+      setShowMinLengthMessage(false)
+      resetSearchResults(null)
+      return
+    }
+
+    if (trimmedQuery.length < MIN_SEARCH_LENGTH) {
       setShowMinLengthMessage(true)
-      setFilteredArticles([])
-      setTotalResults(0)
-      setHasMoreResults(false)
-      setError(null)
+      resetSearchResults(null)
       return
     }
 
     setShowMinLengthMessage(false)
+    setCurrentPage(0) // Reset to first page on new search
+    setError(null) // Clear any previous errors
 
-    if (trimmedQuery.length >= MIN_SEARCH_LENGTH) {
-      setCurrentPage(0) // Reset to first page on new search
-      setError(null) // Clear any previous errors
-
-      startTransition(async () => {
-        try {
-          const result = await searchArticlesAction(trimmedQuery, 0, PAGE_SIZE)
-          if (result.success && result.data) {
-            setFilteredArticles(result.data.data)
-            setTotalResults(result.data.meta.pagination.total)
-            setHasMoreResults(result.data.meta.pagination.pageCount > 1)
-          } else {
-            console.error('Search failed:', result.error)
-            setError('Unable to complete search. Please try again.')
-            setFilteredArticles([])
-            setTotalResults(0)
-            setHasMoreResults(false)
-          }
-        } catch (error) {
-          console.error('Error searching articles:', error)
-          setError('An error occurred while searching. Please try again.')
-          setFilteredArticles([])
-          setTotalResults(0)
-          setHasMoreResults(false)
+    startTransition(async () => {
+      try {
+        const result = await searchArticlesAction(trimmedQuery, 0, PAGE_SIZE)
+        if (result.success && result.data) {
+          setFilteredArticles(result.data.data)
+          setTotalResults(result.data.meta.pagination.total)
+          setHasMoreResults(result.data.meta.pagination.pageCount > 1)
+        } else {
+          console.error('Search failed:', result.error)
+          resetSearchResults('Unable to complete search. Please try again.')
         }
-      })
-    } else {
-      setFilteredArticles([])
-      setTotalResults(0)
-      setHasMoreResults(false)
-      setError(null)
-    }
+      } catch (error) {
+        console.error('Error searching articles:', error)
+        resetSearchResults('An error occurred while searching. Please try again.')
+      }
+    })
   }, [debouncedQuery])
 
-  // Function to load more results
-  const loadMoreResults = async () => {
+  // Function to load more results - memoized to prevent unnecessary recreations
+  const loadMoreResults = React.useCallback(async () => {
```

Comment:
Nitpick: `useCallback` instead of `React.useCallback`

Reply:
Addressed